### ISSUE TITLE
Record information of player deaths

### ISSUE THREAD

**Issue by RussellLVP** [LVPlayground/playground#10]
http://trac.sa-mp.nl/lvp/ticket/1575

Jay:

> A new database table with the following fields:
> 
> > PK Id int
> > FK VictimProfileId? int
> > FK KillerProfileId? int
> > VictimPosX float
> > VictimPosY float
> > VictimPosZ float
> > KillerPosX float
> > KillerPosY float
> > KillerPosZ float
> > reasonId int
> > Timestamp DateTime?
> 
> We'll look to add more detailed data in a later version. This data could be extremely useful for future features

**Comment by mblokker** [comment]
With the implementation of the new DamageManager (https://github.com/LVPlayground/playground/commit/76218d6eb5ee14b79d73a445716dca412a72537f) this can be more easily achieved.

New features could be added like tracking the amount of headshots someone does, which bodyparts they hit the most, how many bullets they have fired, how many have hit a player (= accuracy) etc.

Also think of an Assist feature, giving players assist kills. We can give the player that has inflicted the most damage to a player the kill, while giving participating DM'ers an assist. These are all easy (and fun) additions.

In addition, anticheat!

What is mainly needed here is back-end structure: room in the current database tables to record this. If Russell could make this happen, I would happily continue on the above listed features. I think we could add some fields to 'users_mutable': bullets_fired, bullets_hit, assists, a field for each body part (http://wiki.sa-mp.com/wiki/Body_Parts). These are suggestions, maybe there is room for more!

**Comment by RussellLVP** [comment]
We have been tracking all kills and deaths for a few weeks now, and have recorded 160,884 kills and 35,267 deaths so far. In addition, we've also recorded 5,714,948 weapon hits, growing with quite a lot every day. 

What we don't yet store are the numbers @mblokker suggests in his comment. This would require us to listen to several new frequently called callbacks— let's implement that when the lag problem has been solved.
